src/java/main/org/apache/zookeeper/server/NIOServerCnxn.java

```diff
@@ -716,7 +716,12 @@ public void process(WatchedEvent event) {
         // Convert WatchedEvent to a type that can be sent over the wire
         WatcherEvent e = event.getWrapper();
 
-        sendResponse(h, e, "notification");
+        try {
+            sendResponse(h, e, "notification");
+        } catch (IOException ex) {
+            LOG.debug("Problem sending to " + getRemoteSocketAddress(), ex);
+            close();
```

Comment:
this is an interesting change -- how/when were we closing the connection before?